assignment3/graph.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:io';
+import 'util.dart';
+import 'functions.dart';
+
+class Graph {
+  int nodeId = 0;
+  String nodeName = "";
+  Set<int> immediateChildren = {};
+  Set<int> immediateParent = {};
+
+  Map<dynamic, dynamic> additionalDetails = {};
+  static Set<int> uniqueId = {};
+  static Map<int, Graph> allExistingNodes = {};
+
+  Graph({required this.nodeId, required this.nodeName});
+
+  static bool cyclicityCheck(int node, int parent, int child) {
+    Set<int> visited = {};
+
+    Set<int> allNodes = cyclicityHelper(parent, visited);
+    if (allNodes.contains(child)) {
+      print("dependency cant be creatd");
+      return true;
+    } else
+      return false;
+  }
+
+  static void getImmediateParents(int nodeId) {
```

Comment:
Why are these static? These data points should be accessible through child, right?

Reply:
Same concern for below methods as well. please check

Reply:
this function is a helper function which will be called after the valid node id is entered